src/or/dirvote.c

```diff
@@ -2878,12 +2878,14 @@ dirvote_recalculate_timing(const or_options_t *options, time_t now)
 
 /**
  * Entry point: Take whatever voting actions are pending as of <b>now</b>.
+ *
+ * Return the time at which the next action should be taken.
  */
-void
+time_t
 dirvote_act(const or_options_t *options, time_t now)
 {
   if (!authdir_mode_v3(options))
-    return;
+    return TIME_MAX;
```

Comment:
I don't think -1 means "stop forever"; it means "try again".  TIME_MAX seemed like a reasonable way to say "never".

Reply:
True, sorry `PERIODIC_EVENT_NO_UPDATE = -1` means don't update last timestamp and try again next period.

Reply:
But then we might trigger this `BUG()`:

```
+  if (BUG(next == TIME_MAX)) {
+    return 3600;
+  }
```

... which in theory should never be triggered since the callback should never run if `authdir_mode_v3()` returns false.

Reply:
cd45c16d9a3eb85 has some comments to try to explain this.  Please let me know if that isn't what you had in mind.